File: ui/src/models/dto/RecipeDto.ts
import type Ingredient from "../domain/Ingredient.ts";
import type PreparationStep from "../domain/PreparationStep.ts";
import type { PreparationTime } from "../domain/PreparationTime.ts";

export default interface RecipeDto {
    id: number;
    title: string;
    description: string;
    imageName: string | null;
    numServings: number;
    preparationTime: PreparationTime;
    author: {
        userId: number;
        username: string;
    };
    ingredients: Ingredient[];
    preparationSteps: PreparationStep[];
}

File: app/src/main/java/dev/blaauwendraad/recipe_book/web/model/RecipeDto.java
package dev.blaauwendraad.recipe_book.web.model;

import dev.blaauwendraad.recipe_book.service.model.PreparationTime;
import jakarta.annotation.Nullable;
import java.util.List;

public record RecipeDto(
        Long id,
        String title,
        @Nullable String description,
        @Nullable String imageName,
        Integer numServings,
        PreparationTime preparationTime,
        @Nullable RecipeAuthorDto author,
        List<IngredientDto> ingredients,
        List<PreparationStepDto> preparationSteps) {}
